pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -681,6 +678,22 @@ void messageReceived(MessageIdData messageId, ByteBuf headersAndPayload, ClientC
                 }
             });
         }
+
+        if (receiveListener != null) {
```

Comment:
thought about that, should I document that it should not block and execute it from the IO thread?